pkgs/top-level/python-packages.nix

```diff
@@ -5212,7 +5210,7 @@ in {
   protobuf = callPackage ../development/python-modules/protobuf {
     disabled = isPyPy;
     doCheck = !isPy3k;
-    protobuf = pkgs.protobuf3_8;
+    protobuf = pkgs.protobuf3_12;
```

Comment:
I think this was added for tensorflow, because it requires protobuf==3.8 https://github.com/NixOS/nixpkgs/pull/81278 https://github.com/tensorflow/tensorflow/issues/35573
Could you please add an override for tensorflow to use protobuf3_8?

Reply:
That's not ok because then one can get a closure with multiple versions of the protobuf package. This is unsolvable with Python and a reason why we do *not* always use the latest version of libraries but instead versions that keep most things working.

Reply:
I understand the reason, but this makes it impossible to package software that requires newer python-protobuf into nixpkgs.

Reply:
Yes, in case of *libraries*, you are correct. We need to balance this. Thus, that also means, if one or a few packages are extremely slow updating, we may need to reconsider whether they should be part of Nixpkgs.

Reply:
TensorFlow [requires](https://github.com/tensorflow/tensorflow/blob/v2.2.0/tensorflow/tools/pip_package/setup.py#L63) `protobuf >= 3.8.0`, not `==`.

#81278 bumped this line from 3.7 to 3.8, but doesn’t seem to indicate that a later version wouldn’t have worked. tensorflow/tensorflow#35573 appears to be an issue with a specific binary distribution.

I see no reason to think it wouldn’t work if everything consistently used the current stable version. Why not `protobuf = pkgs.protobuf`?

Reply:
> I see no reason to think it wouldn’t work if everything consistently used the current stable version. Why not protobuf = pkgs.protobuf?

That sounds good to me. Care to send a PR trying that for tensorflow?

Reply:
@bhipple It would have no effect without this PR, because we still have `protobuf = protobuf3_8` in `master`.

Reply:
My take is: let's get this into master and fix breakages that might happen.